packages/@lwc/synthetic-shadow/src/faux-shadow/lwc-host.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty } from '@lwc/shared';
+import { setAttribute } from '../env/element';
+
+/*
+ * Using a simple custom attribute is faster based on perf results:
+ * Setting attribute: https://jsperf.com/attribute-vs-data-attribute/1
+ * matches(selector): https://jsperf.com/matches-attribute-vs-data-attribute/1
+ * querySelector(): https://jsperf.com/query-by-attribute-vs-data-attribute/1
+ **/
+export const HostTokenAttributeName = 'lwc-host';
```

Comment:
I think it should be a namespaced attribute: `lwc:host`

Reply:
For a query like `[lwc\\:host] *` it's typically more performant cross-browser to use a class rather than an attribute (because [it's more likely to fall into the Bloom filter](https://salesforce.quip.com/vIxDAVXHBCsv)). Can this be a class rather than an attribute?